internal/runners/builds/download.go

```diff
@@ -116,7 +116,19 @@ func (d *Download) downloadArtifact(artifact *artifact.Artifact, targetDir strin
 		return errs.Wrap(err, "Could not parse artifact URL %s.", artifact.URL)
 	}
 
-	downloadPath := filepath.Join(targetDir, path.Base(artifactURL.Path))
+	// Determine an appropriate basename for the artifact.
+	// Most platform artifact URLs are just "artifact.tar.gz", so use "<name>-<version>.<ext>" format.
+	// Some URLs are more complex like "<name>-<hash>.<ext>", so just leave them alone.
+	basename := path.Base(artifactURL.Path)
```

Comment:
I might be missing something here, so forgive me if I am, but what does the URL have to do with the naming aside from the file extension? We resolved the artifact name by parsing the build plan so we shouldn't care what the URL is.

Reply:
Docker images have weird names like "Docker Image (more-descriptive-name-hash).tar.gz". However, the URL name only has "more-descriptive-name-hash.tar.gz", which is what we want.